src/plugins/ui_stateful/header_positions.ts

```diff
@@ -47,6 +47,7 @@ export class HeaderPositionsUIPlugin extends UIPlugin {
       case "UNGROUP_HEADERS":
       case "GROUP_HEADERS":
       case "CREATE_SHEET":
+      case "SET_FORMATTING":
```

Comment:
We can update the style of a cell without `SET_FORMATTING`, `SET_FORMATTING` is just a wrapper around `UPDATE_CELL`.

You should change the `UPDATE_CELL` of this switch. It should be
```js
      case "UPDATE_CELL":
        if ("content" in cmd) {
          this.headerPositions = {};
          this.isDirty = true;
        } else {
          this.headerPositions[cmd.sheetId] = this.computeHeaderPositionsOfSheet(cmd.sheetId);
        }
```

if the content of the cell changed, we need to mark the headers sizes as dirty while waiting for the evaluation. Otherwise if the style/format changed we only recompute the header sizes for this sheet

Reply:
Hello @hokolomopo,

When we change the format of content filtered in a data filter, It is not computing header positions correctly, Which results in keeping rows hidden inside the data filter and weird behavior between two plugins.

[Check out the failing test case here](https://github.com/odoo/o-spreadsheet/blob/17.0/tests/sheet/sheetview_plugin.test.ts#L846-#L861)

Should we keep the format to compute the header position after evaluation?

Reply:
I have implemented the suggested changes, Thanks :)